src/structure/grid.ts

```diff
@@ -478,6 +489,41 @@ export class ViewersGrid {
         return this._resolvedStructures[index];
     }
 
+    private _getAtomProperties(): Record<string, Property> {
+        const numberProperties = filter(this._properties, (p) =>
+            Object.values(p.values).every((v) => typeof v === 'number')
+        );
+        const atomProperties = filter(numberProperties, (p) => p.target === 'atom');
+        return atomProperties;
+    }
```

Comment:
Instead of storing all properties & then looking for properties containing number & with `target =  atom`, I would only store the ones we need in `this->_properties` and do this filtering in the constructor

Reply:
Fixed